qt/python/mantidqtinterfaces/mantidqtinterfaces/Muon/GUI/Common/phase_table_widget/phase_table_presenter.py

```diff
@@ -217,60 +181,60 @@ def handle_calculate_phase_table_clicked(self):
+        self.model.calculation_thread = self.create_phase_table_calculation_thread()
+        self.model.calculation_thread.threadWrapperSetUp(
             self.handle_phase_table_calculation_started,
             self.handle_phase_table_calculation_success,
             self.handle_phase_table_calculation_error,
         )
-        self.calculation_thread.start()
+        self.model.calculation_thread.start()
 
     def handle_phase_table_changed(self):
         """Handles when phase table is changed, recalculates any existing phasequads"""
         self.disable_editing_notifier.notify_subscribers()
         self.view.disable_widget()
-        current_table = self.context.phase_context.options_dict["phase_table_for_phase_quad"]
+        current_table = self.model.phase_context.options_dict["phase_table_for_phase_quad"]
         new_table = self.view.get_phase_table()
         if new_table == current_table:
             return
-        self.context.phase_context.options_dict["phase_table_for_phase_quad"] = new_table
+        self.model.phase_context.options_dict["phase_table_for_phase_quad"] = new_table
 
         # Update the table stored in each phasequad
-        self.context.group_pair_context.update_phase_tables(new_table)
-        self.context.update_phasequads()  # Updates phasequads
+        self.model.group_pair_context.update_phase_tables(new_table)
+        self.model.context.update_phasequads()  # Updates phasequads
         self.calculation_finished_notifier.notify_subscribers()
         self.view.enable_widget()
         self.enable_editing_notifier.notify_subscribers()
 
     """=============== Phasequad methods ==============="""
 
     def calculate_phasequad(self):
-        self.context.group_pair_context.add_phasequad(self._phasequad_obj)
-        self.context.calculate_phasequads(self._phasequad_obj)
+        self.model.group_pair_context.add_phasequad(self.model.phasequad)
+        self.model.context.calculate_phasequads(self.model.phasequad)
 
-        self.phasequad_calculation_complete_notifier.notify_subscribers(self._phasequad_obj.Re.name)
-        self.phasequad_calculation_complete_notifier.notify_subscribers(self._phasequad_obj.Im.name)
+        self.phasequad_calculation_complete_notifier.notify_subscribers(self.model.phasequad.Re.name)
+        self.phasequad_calculation_complete_notifier.notify_subscribers(self.model.phasequad.Im.name)
 
     def remove_last_row(self):
         if self.view.num_rows() > 0:
             name = self.view.get_table_contents()[-1]
             self.view.remove_last_row()
-            for phasequad in self.context.group_pair_context.phasequads:
+            for phasequad in self.model.group_phasequads:
                 if phasequad.name == name:
                     self.add_phasequad_to_analysis(False, False, phasequad)
-                    self.context.group_pair_context.remove_phasequad(phasequad)
+                    self.model.group_pair_context.remove_phasequad(phasequad)
```

Comment:
Is it worth wrapping this in a `remove_phasequad` method on the model, as it seems to be used in two places?